File: src/main/java/com/pfe/elearning/MeetingSchedule/dto/MeetingScheduleResponse.java
package com.pfe.elearning.MeetingSchedule.dto;

import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.*;
import java.time.Duration;
import java.time.LocalDateTime;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class MeetingScheduleResponse {

    private Integer id;
    private String meetingTitle;
    private Integer durationMinutes;
 //   @JsonFormat(pattern = "yyyy-MM-dd")
  //  private LocalDateTime dateTime;
  private String dateTime;

    private String meetingUrl;
    private String meetingDescription;
    private Integer courseId; // Course ID
    private String courseTitle; // Course title
    private String instructorFullName;
    private Integer lessonId; // Lesson ID
    private String lessonTitle; // Lesson title
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
}

File: src/app/models/MeetingSchedule/MeetingSchedule-response.ts
export interface MeetingScheduleResponse {
  id: number;
  meetingTitle: string;
  durationMinutes: number; // Duration in minutes
  dateTime: string; // Date and time in ISO 8601 format
  meetingUrl: string;
  meetingDescription: string;
  instructorFullName: string;
  courseId: number;
  courseTitle: string;
  lessonId: number;
  lessonTitle: string;
  createdAt: string; // Date and time of creation in ISO 8601 format
  updatedAt: string; // Date and time of last update in ISO 8601 format
  endTime?: string; // End time in ISO 8601 format

}
